.github/actions/build-docker/action.yml

```diff
@@ -0,0 +1,82 @@
+name: 'Docker Build'
+description: 'Builds `addons-server` docker image'
+inputs:
+  no_cache:
+    required: false
+    description: "Build fresh image without cache"
+    default: "false"
+  push:
+    required: false
+    description: "Build and push image to registry (cannot be used together with load)"
+    default: "false"
+  pull:
+    required: false
+    description: "Build and push image to registry"
+    default: "false"
+  load:
+    required: false
+    description: "Build and load image to local docker daemon. (cannot be used together with push)"
+    default: "false"
+  cache_dir:
+    required: false
+    description: "Docker cache directory"
+    default: /tmp/cache/docker-ci
+  python_version:
+    required: false
+    description: "Python version"
+    default: "3.10"
+
+outputs:
+  tags:
+    description: "The Docker tags for the image"
+    value: ${{ steps.meta.outputs.tags }}
+
+runs:
+  using: "composite"
+  steps:
+    # Setup docker to build for multiple architectures
+    - name: Set up QEMU
+      uses: docker/setup-qemu-action@v3
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v1
+      with:
+        version: latest
+        buildkitd-flags: --debug
+
+    # Login to a registry to push the image
+    - name: Login to GitHub Container Registry
+      uses: docker/login-action@v3
+      with:
+        registry: ghcr.io
```

Comment:
We're not using `ghcr.io` at the moment are we?  I thought we used dockerhub?  Is the intent to move away from dockerhub or duplicate builds? (why, in either case)

Reply:
I've set it up with ghcr.io for now for 2 reasons:
1. don't pollute our main registry we actually use for production
2. it is easy to setup because auth is implicit.

We can switch this action to use dockerhub if we want,it just means we would need to add credentials as secrets in the repo. Didn't seem important yet to do so.

Reply:
My instinct is that we should stick with a single integration - either ghcr.io or dockerhub - but I don't have an opinion right now which. @diox do you?

Reply:
I think long term yes, just don't want to block this PR on decisions that are not urgent to make and we should carefully consider.

Reply:
Do we have to push the image at all ?

Reply:
If you want to be able to run and test it outside of CI then yes. I did this several times when working on implementing this PR and can see myself doing it very regularly. You build an Image and want to run that image locally to get "exactly" what the CI built in your environment.

It also has huge implications for this [decision](https://mozilla-hub.atlassian.net/wiki/spaces/FDPDT/pages/564627681/Everything+on+Master+must+be+Deployable)

Reply:
Though the default behaviour on the `verify-docker-image` is to [load](https://github.com/mozilla/addons-server/pull/21930/files#diff-c54769fa60b5344b018413c90790e136a0ab1d23ef81b6dfaf68135c220317a2R19) instead of push. You can build it and test it, but it won't push.

If you want to push an image you can run the workflow to push the image. I imagine we may find different use cases for pushing but for now this is flexible.

Reply:
Personally, I don't care that much if we're building to different registry in this PR, but I do think there should be a follow-up issue to consolidate to one or the other.

Reply:
Done. https://github.com/mozilla/addons-server/issues/21940

Reply:
Both registries have potential costs involved so we shouldn't push to both if not actually required IMHO.

Reply:
I think we are answering the wrong question right now. Do we want to block this PR on that decision? Or is it fine to merge and make the registry decision in a separate iteration. In order to push anything to ghcr you'd have to manually initiate a job, so I think the risk is low, but if you'd rather keep it all on dockerhub for now then we need to set the credentials in the repo and I can update the PR to login there.

I would recommend we keep it as is and tackle the registry question in the linked ticket.

Reply:
dockerhub credentials were added so we can proceed with 1 registry.